src/components/BackOffice/ContractsDashboard/ContractorsListItems.tsx

```diff
@@ -1,52 +1,68 @@
-import ContractorListItem from './ContractorListItem'
-import { Button } from '../../Form'
-
 import Contract from '@/root/src/models/contract'
+import ContractorListItem from './ContractorListItem'
+import WarningInfoBox from '../../Template/WarningInfoBox'
 
 interface ContractorsListItemsProps {
-  filteredContracts: Contract[]
-  setPageNumber: (pageNumber: number) => void
-  pageNumber: number
-  totalPages: number
+  contracts: Contract[]
 }
 
-const ContractorsListItems = ({
-  filteredContracts,
-  setPageNumber,
-  pageNumber,
-  totalPages,
-}: ContractorsListItemsProps) => {
-  if (filteredContracts === null || filteredContracts?.length === 0) {
-    return <></>
+const ContractorsListItems = ({ contracts }: ContractorsListItemsProps) => {
+  const relevantContracts = contracts?.filter((c) => c.terminationDate > '2020')
+
+  const contractorNames = new Set<string>(
+    relevantContracts.map((contract) => contract.contractorName)
+  )
+  const contractorReferences = new Set<string>(
+    relevantContracts.map((contract) => contract.contractorReference)
+  )
+
+  const contractorNamesAndReferences = [...contractorNames].map(
+    (name, index) => ({
+      contractorName: name,
+      contractorReference: [...contractorReferences][index],
+    })
+  )
+
+  const contractorNamesAndReferencesSortedByContractorName = contractorNamesAndReferences.sort(
+    (a, b) => a.contractorName.localeCompare(b.contractorName)
+  )
```

Comment:
Did you have an issue with duplicate names and references? This could be a fair bit simpler if you don't need to filter duplicates out